pkg/3scale/amp/operator/backend_reconciler.go

```diff
@@ -154,6 +154,60 @@ func (r *BackendReconciler) Reconcile() (reconcile.Result, error) {
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+	if r.apiManager.Spec.Backend.ListenerSpec.Hpa.Enabled || r.apiManager.Spec.Backend.WorkerSpec.Hpa.Enabled {
```

Comment:
so basically this means that we can have apicast in non-hpa while backends in hpa but we can't have for example a worker in hpa but listener in non-hpa mode right?

Reply:
also, if I want to delete my hpas - I set it to false, the deletion is triggered in the r.ReconcileHPA, correct me if this is incorrect but this means in the event where I want to have my apicast hpa on, but turn off the backend hpa - it won't work from this reconciler but the deletion of backend hpas would be more of a side effect of apicast reconciler going into the ReconcileHPA func.

Reply:
first point makes sense, I would need to check again before calling r.ReconcileHpa if I want the backend components to be individually enabled.

I think your second point also makes sense,  if both backend components are false then we never hit r.ReconcileHpa, and never hit the delete logic.  

Need to have a think about it.

Reply:
the check is not needed as we check if enabled in r.ReconcileHPA,   its only needed to set the status to nil so a seperate check on the status is needed.

Reply:
done